llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -1344,8 +1264,10 @@ bool NVPTXDAGToDAGISel::tryLDGLDU(SDNode *N) {
       return false;
     SDValue Ops[] = { Addr, Chain };
     LD = CurDAG->getMachineNode(*Opcode, DL, InstVTList, Ops);
-  } else if (TM.is64Bit() ? SelectADDRri64(Op1.getNode(), Op1, Base, Offset)
-                          : SelectADDRri(Op1.getNode(), Op1, Base, Offset)) {
+  } else {
+    TM.is64Bit() ? SelectADDRri64(Op1.getNode(), Op1, Base, Offset)
+                 : SelectADDRri(Op1.getNode(), Op1, Base, Offset);
```

Comment:
That used to be the condition for `if` but does not do much now.
Did you intend to make it an assertion?

Reply:
No, Base and Offset are references which these functions will place the correct values into in order to return them.

Reply:
Right. I could not use new C++ features those days. These days I'd just return an optional tuple of {base, offset}.

I'd switch to a regular `if` here. Ternary suggests that we're interested in the return value of SelectAddr*